models/convert/transactions.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package convert
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+// EventsToTransactions processes a slice of events into a map of Rosetta transactions. It ensures that
+// each operation included in transactions has its related operation IDs set accordingly, and maps transactions
+// by their ID.
+func EventsToTransactions(ee []flow.Event, withdrawal string) (map[string]*object.Transaction, error) {
```

Comment:
I think this function is still not really where we want to go. I would suggest that we want to have a `EventToOperation` function, which returns `object.Operation, bool, error`. If the type is unknown, it return false and no operation. When the type is known (at the moment `withdrawal` or `deposit`, but later also others), it tries to convert and errors in case there are issues.

Within the calling function, we then just iterate through all the events in the block and try to convert them. For the transaction, we only try to convert those that have the right transaction ID.

Reply:
> When the type is known (at the moment withdrawal or deposit, but later also others), it tries to convert and errors in case there are issues.

WDYM by that? Currently a withdrawal and a deposit operation would look exactly the same, besides their `Type` field that lets us know whether it is a deposit or withdrawal. In what case would we return no error but a false boolean, then? 🤔 

Also, we agree that reverting the amount for withdrawals would be done in the retriever then, right? Since this function has no clue what the exact event type for a withdrawal is.